kubespawner/reflector.py

```diff
@@ -152,16 +184,26 @@ def _list_and_update(self):
 
         Overwrites all current resource info.
         """
-        initial_resources = getattr(self.api, self.list_method_name)(
-            self.namespace,
-            label_selector=self.label_selector,
-            field_selector=self.field_selector,
-            _request_timeout=self.request_timeout,
-            _preload_content=False,
-        )
+        initial_resources = None
+        if self.omit_namespace:
+            initial_resources = getattr(self.api, self.list_method_name)(
```

Comment:
Can probably avoid this duplication with something like:

```python
kwargs = dict(
    label_selector=self.label_selector,
    field_selector=self.field_selector,
    _request_timeout=self.request_timeout,
    _preload_content=False,
)
if not self.omit_namespace:
    kwargs["namespace"] = self.namespace

initial_resources = getattr(self.api, self.list_method_name)(**kwargs)
```

which I see you are already doing below with `watch_args`

Reply:
Done!